src/test/java/org/opensearch/ad/rest/AnomalyDetectorRestApiIT.java

```diff
@@ -1504,4 +1508,393 @@ public void testValidateAnomalyDetectorWithWrongCategoryField() throws Exception
         );
 
     }
+
+    public void testSearchTopAnomalyResultsWithInvalidInputs() throws IOException {
+        String indexName = randomAlphaOfLength(10).toLowerCase();
```

Comment:
Index name must not contain some special chars like "#", "|", and must not start with "-", "+" or "_".
Not sure if the `randomAlphaOfLength` will always return valid index name. How about we use some concrete index name to avoid flaky test?

Reply:
And same concern for detector name as Amit added regex checking detector name. @amitgalitz can you help check if we have such potential flaky part?

Reply:
Good catch. But, followed the stack of calls, and seems `randomAlphaOfLength` will only return `a-zA-Z`, since it's coming from this generator:
```
public class AsciiLettersGenerator extends CodepointSetGenerator {
  private final static char [] CHARS = 
      ("abcdefghijklmnopqrstuvwxyz" + 
       "ABCDEFGHIJKLMNOPQRSTUVWXYZ").toCharArray();

  public AsciiLettersGenerator() {
    super(CHARS);
  }
}
```

Reply:
+1 on what Tyler mentioned, doesn't create anything flaky.

Reply:
Cool, thanks for confirming that